daft/runners/pyrunner.py

```diff
@@ -107,27 +108,118 @@ def wait(self) -> None:
         pass
 
 
-class PyActorPool:
-    initialized_stateful_udfs_process_singleton: dict[str, UserProvidedPythonFunction] | None = None
+@dataclass
+class PyRunnerResources:
+    num_cpus: float
+    gpus: dict[str, float]
+    memory_bytes: int
 
-    def __init__(
-        self,
-        pool_id: str,
-        num_actors: int,
+    def __repr__(self) -> str:
+        return f"PyRunnerResources(num_cpus={self.num_cpus}, gpus={self.gpus}, memory_bytes={self.memory_bytes})"
```

Comment:
I think dataclasses have a default repr that looks pretty much like this already